python/metatensor-operations/metatensor/operations/_dispatch.py

```diff
@@ -527,6 +527,35 @@ def int_array_like(int_list: List[int], like):
         raise TypeError(UNKNOWN_ARRAY_TYPE)
 
 
+def indices_like(shape: List[int], like):
+    """
+    Creates a tensor of shape ``shape`` filled with ints that
+    enumerate the indices of the entries in the array, e.g.
+    ``shape = (3,2)`` returns ``[[0,0],[0,1],[1,0],[1,1],[2,0],[2,1]]``.
+
+    If the backend is torch and the device is "meta",
+    the device is set to "cpu". This is useful in case where
+    we create labels for a block that is on the meta device.
+    In that case, `int_list` are the labels, and `like` are the block
+    values.
```

Comment:
I'm not sure I like special cases like this inside `_dispatch`. IMO this should be handled at the caller level

Reply:
This was just copied from int_array_like... not clear why it was there, happy to take it out

Reply:
ah, ok =). I'll see about this one then